### DISCUSSION TITLE
Portal Support for Dropdown Menu

### DISCUSSION THREAD

**Discussion by BrianHung** [tailwindlabs/headlessui#1346] [Ideas]
It would be nice if [dropdown menus](https://headlessui.dev/react/menu) could support portals like [panels](https://headlessui.dev/react/popover#positioning-the-panel).

A use-case I've ran into a dropdown menu being cut-off by an ancestor's overflow-x hidden; by moving the rendering root to the body, the overflow-x: hidden wouldn't affect the rendering of the menu.

A similar library to headless that supports this is https://www.radix-ui.com/docs/primitives/components/dropdown-menu

**Edit**: Digging a bit more into headlessui's codebase, there _is_ a portal element that's used for dialogs: https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/portal/portal.tsx

Using this correctly still requires a positioning library e.g. `react-popover` though. Radix itself uses an internal positioning component in addition to their portal: https://github.com/radix-ui/primitives/tree/main/packages/react/popper/src

**Comment by ycs77** [comment]
Hi! I create a package [Headless UI Float](https://github.com/ycs77/headlessui-float), that can be easy to position floating Headless UI elements using Floating UI (is new version Popper.js), support Transition & Portal.

If you need to float Headless UI element can try it out 😊

**Reply by sb8244** [reply]
Just tried this out and had it working exactly like I wanted within 5 minutes! Awesome work.

**Reply by SteffeyDev** [reply]
Lib works perfectly, thanks! would still love to see this built in

**Reply by DavydenkovM** [reply]
Btw about the headless-ui Float  package - there's a prop `strategy="fixed"` - it also could help you with cut-off dropdowns instead of Portals (it helped me with similar issue)

like

```
<Float strategy="fixed">
            <Popover.Button ref={setReferenceElement}>
              {children}
            </Popover.Button>

            <div class="p-4">
              ...
            </div>
</Float>
```

**Reply by zachblume** [reply]
Huge shoutout, `<Float portal>...</Float>` did the trick for this issue for me. Nicely maintained library
